Sources/Utility/ArgumentParser.swift

```diff
@@ -433,21 +523,30 @@ public final class ArgumentParser {
         kind: T.Type,
         usage: String? = nil
     ) -> OptionArgument<T> {
-        let arg = OptionArgument<T>(name: option, shortName: shortName, usage: usage)
-        options.append(AnyArgument(arg))
-        return arg
+        precondition(!optionArguments.contains(where: { $0.argument.name == option }), "Can not define an option twice")
```

Comment:
We should turn this into assert, this should be easily catchable in debug builds.